pkg/gameserversets/controller.go

```diff
@@ -356,6 +356,7 @@ func (c *Controller) syncGameServerSet(ctx context.Context, key string) error {
 
 	if numServersToAdd > 0 {
 		if err := c.addMoreGameServers(ctx, gsSet, numServersToAdd); err != nil {
+			c.workerqueue.EnqueueAfter(gsSet, 1*time.Second)
```

Comment:
👀 this looks like a good find.  I'm kinda wondering why we didn't return the error here in the first place.

Rather than re-enqueue, the canonical way to re-queue the item, is to return the error after the log, which gets handled by the workerqueue as a retry mechanism.

Since we're in here, we should also do the same at line 365/366 because if the delete fails, we should also retry.

Nice catch!